apps/studio/src/server/modules/resource/resource.router.ts

```diff
@@ -200,19 +202,40 @@ export const resourceRouter = router({
       } else {
         query = query.where("Resource.parentId", "=", String(resourceId))
       }
-
       const result = await query.execute()
-      if (result.length > limit) {
+      return {
```

Comment:
comment/nit: this seems like unnecessary optimisation (actually it's less performant so not really an optimisation too (`slice` recreates the array, `pop` mutates)) + less readable. Suggest sticking with previous implementation.